buildsystem/DependencyFetch.cmake

```diff
@@ -0,0 +1,66 @@
+# Copyright 2017-2017 the openage authors. See copying.md for legal info.
+
+# cmake external project wrapper
+
+
+
+function(fetch_project)
+	set(KW_ARGS
+		NAME
+		LOCATION
+		TAG
+	)
+
+	cmake_parse_arguments(PROJ "" "${KW_ARGS}" "" ${ARGN})
+
+	if(PROJ_NAME STREQUAL "")
+		message(FATAL_ERROR "no project name given")
+	endif()
+
+	set(PROJ_DIR "${CMAKE_BINARY_DIR}/${PROJ_NAME}-external")
+	set(PROJ_DL_DIR "${PROJ_DIR}/dl")
+	set(PROJ_SRC_DIR "${PROJ_DIR}/source")
+	set(PROJ_BIN_DIR "${PROJ_DIR}/bin")
+	set(PROJ_STAMP_DIR "${PROJ_DIR}/stamp")
+
+	# CLion creates this file for multiple projects
+	file(REMOVE "${PROJ_DL_DIR}/CMakeCache.txt")
+
+	# create the ExternalProject configuration file
+	# for the nested cmake call.
+	configure_file(
+		"${BUILDSYSTEM_DIR}/templates/ExternalFetch.cmake.in"
+		"${PROJ_DL_DIR}/CMakeLists.txt"
+		@ONLY
+	)
+
+	# run cmake to "configure" the external project.
+	# this prepares the download that will be done in the next step.
+	execute_process(COMMAND
+		${CMAKE_COMMAND}
+		"-G${CMAKE_GENERATOR}"
+		"-DCMAKE_MAKE_PROGRAM:FILE=${CMAKE_MAKE_PROGRAM}"
```

Comment:
As we've established this function just downloads the project, it should be alright to let CMake use the defaults for everything: the generator, and `CMAKE_MAKE_PROGRAM`.

Reply:
That's right. The 'download project' should not be needed anywhere. What generator does it use on windows then?

Reply:
The Windows build for CMake uses the latest available "Visual Studio" generator.
The package from cygwin uses "Unix Makefiles".
I don't remember for MSYS/MinGW package(s) - my guess: "MSYS Makefiles" and "MinGW Makefiles" respectively.

Reply:
What's the problem with passing through customizations?

Reply:
There's no problem. The comment was just to reduce boilerplate code.

Reply:
Okay, so what do? Should we keep it or use the defaults? I'd go for passing the configuration, then customizations will propagate and not produce an unexpected and uninfluenceable subproject buildsystem.

Reply:
Fair point. We can keep it.
It might help someone later. 👍